source/common/upstream/maglev_lb.h

```diff
@@ -54,8 +58,29 @@ class MaglevLoadBalancer : public ThreadAwareLoadBalancerBase {
 
 private:
   // ThreadAwareLoadBalancerBase
-  HashingLoadBalancerSharedPtr createLoadBalancer(const HostVector& hosts) override {
-    return std::make_shared<MaglevTable>(hosts, table_size_);
+  HashingLoadBalancerSharedPtr createLoadBalancer(const HostSet& host_set) override {
+    // Note that we only compute global panic on host set refresh. Given that the runtime setting
+    // will rarely change, this is a reasonable compromise to avoid creating extra LBs when we only
+    // need to create one per priority level.
+    const bool has_locality =
+        host_set.localityWeights() != nullptr && !host_set.localityWeights()->empty();
+    if (isGlobalPanic(host_set)) {
```

Comment:
In looking at the code duplication here I wonder if there is a way to still have the panic computation happen generally, and then we just get passed either healthy or not healthy host sets, even if Ketama doesn't operate on a per locality basis? Might not be worth it but just throwing it out there. Mainly just wondering the reasoning behind deciding to duplicate vs. generalize.

Either way please check to make sure we have coverage on all these paths.

Reply:
Host sets include both healthy/unhealthy, it's only individual host lists or hosts per locality that can be split out like that. The rationale for the current structure is that I didn't want to complicate Ketama with dealing with localities.

We have full coverage, see https://37894-65214191-gh.circle-artifacts.com/0/build/envoy/generated/coverage/coverage.bazel-out_k8-dbg_bin_source_common_upstream__virtual_includes_maglev_lb_lib_common_upstream_maglev_lb.h.html and https://37894-65214191-gh.circle-artifacts.com/0/build/envoy/generated/coverage/coverage.bazel-out_k8-dbg_bin_source_common_upstream__virtual_includes_ring_hash_lb_lib_common_upstream_ring_hash_lb.h.html.

Reply:
OK fair enough. I guess we can revisit in the future if we find we keep duplicating this logic to more load balancers.